test/distributed/checkpoint/test_state_dict.py

```diff
@@ -576,6 +576,17 @@ def set_extra_state(self, state):
         self.assertEqual(model.state_dict()["u1._extra_state"], "MyState")
         self.assertEqual(model.state_dict(), get_model_state_dict(ddp_model))
 
+    @with_comms
+    @skip_if_lt_x_gpu(1)
+    def test_non_persistent_buffers(self) -> None:
+        model = CompositeParamModel(device=torch.device("cuda"))
+        model.register_buffer(
+            "dont_save_me", torch.rand(100, device="cuda"), persistent=False
+        )
+        ddp_model = DDP(copy.deepcopy(model))
+        set_model_state_dict(ddp_model, get_model_state_dict(ddp_model))
```

Comment:
IIUC, `set_model_state_dict(module, get_model_state_dict(module))` should be a no-op. Is this just testing that `set_model_state_dict()` does not error?